net-p2p/cardano-db-sync/files/cardano_db_sync.in

```diff
@@ -0,0 +1,161 @@
+# cardano_db_sync_enable:          Set to YES to enable cardano_db_sync.
+#                               Default: "NO"
+#
+# cardano_db_sync_net:             A network name to connect to.
+#                               Default: "mainnet"
+#
+# cardano_db_sync_home:            An absolute path to the daemon home directory.
+#                               The directory will be created if not exists.
+#                               Default: "/var/db/cardano_db_sync"
+#
+# cardano_db_sync_pgpass:          An absolute path to the PostgreSQL ".pgpass" connection file. 
+#                               Default: "${cardano_db_sync_home}/${cardano_db_sync_net}-configs/.pgpass"
+#
+# cardano_db_sync_cnode_socket:    An absolute path to the cardano-node socket file.
+#                               Default: "/var/db/cardano_node/jail/socket/cardano-node.sock"
+#
+# cardano_db_sync_config:          An absolute path to the db-sync configuration JSON file.
+#                               Default: "${cardano_db_sync_home}/${cardano_db_sync_net}-configs/db-sync-config.json"
+# 
+# cardano_db_sync_schema:          An absolute path to the db-sync schema directory.
+#                               Default: "/usr/local/share/cardano-db-sync/schema"
+#
+# cardano_db_sync_state:          An absolute path to the db-sync ledger-state directory.
+#                               If it does not exist, a new one will be created in this path.
+#                               Default: "${cardano_db_sync_home}/${cardano_db_sync_net}-state"
+#
+# cardano_db_sync_flags:           Any additional command line flags to pass to cardano_db_sync.
+#                               Default: ""
+#
+
+. /etc/rc.subr
+
+name=cardano_db_sync
+desc="cardano_db_sync daemon"
+rcvar=cardano_db_sync_enable
+command=%%PREFIX%%/libexec/cabal/cardano-db-sync
+
+cardano_deployment_url="https://raw.githubusercontent.com/cardano-bsd-alliance/freebsd-ports-cardano-artifacts/master/cardano-db-sync"
+cardano_config_files="config db-sync-config byron-genesis shelley-genesis alonzo-genesis"
+cardano_networks="mainnet testnet"
+
+start_cmd="${name}_start"
+start_precmd="${name}_prestart"
+stop_cmd="${name}_stop"
+status_cmd="${name}_status"
+fetch_cmd="${name}_fetch"
+
+extra_commands="status fetch"
+
+load_rc_config $name
+: ${cardano_db_sync_enable:=NO}
+: ${cardano_db_sync_net:="mainnet"}
+: ${cardano_db_sync_home:="/var/db/cardano_db_sync"}
+: ${cardano_db_sync_pgpass:="${cardano_db_sync_home}/${cardano_db_sync_net}-configs/.pgpass"}
+: ${cardano_db_sync_cnode_socket:="/var/db/cardano_node/jail/socket/cardano-node.sock"}
+: ${cardano_db_sync_config:="${cardano_db_sync_home}/${cardano_db_sync_net}-configs/db-sync-config.json"}
+: ${cardano_db_sync_schema:="/usr/local/share/cardano-db-sync/schema"}
+: ${cardano_db_sync_state:="${cardano_db_sync_home}/${cardano_db_sync_net}-state"}
+: ${cardano_db_sync_flags:=""}
+
+export PATH="/sbin:/bin:/usr/sbin:/usr/bin:/usr/local/sbin:/usr/local/bin:/root/bin"
```

Comment:
So, i don't know how else to solve this.

export for the .pgpass is needed because the binary will not run without it at all. It needs that environment variable to start.

and the export for PATH, is because when the binary starts, it's looking to execute pqsql binary, to run the .SQL files ( the 
 ones under /share/ folder ).  Is there another way to make sure there is no "pqsql not found" error is resolved?

Reply:
Interesting, the `PATH` is set to `/sbin:/bin:/usr/sbin:/usr/bin` by default in the rc script. So we need to append `/usr/local/bin` to that. I'd recommend running

`/usr/bin/env PATH=$PATH:/usr/local/bin /usr/bin/daemon ...`

instead of just `/usr/bin/daemon ...` in the `start` function. This will modify `PATH` only for the actual `cardano-db-sync` invocation, not the whole rc script.

Reply:
Will test and report back.

Reply:
Oh! This should be even `/usr/bin/env PATH=$PATH:%%LOCALBASE%%/bin` and then replaced with `${LOCALBASE}` using `SUB_LIST`.

Reply:
That seems like a better solution. What about  Default: "/usr/local/share/cardano-db-sync/schema"  Should we 
 modify to %%LOCALBASE%%/share/cardano-db-sync/schema instead?

Reply:
This works! Including the change to the schema folder.